### ISSUE TITLE
Disallow circular routes to the same peer

### ISSUE THREAD

**Issue by joostjager** [lightningnetwork/lnd#4698]
In PR https://github.com/lightningnetwork/lnd/pull/3915 a restriction was introduced to prevent incoming htlcs from being forwarded out through the same channel.

This doesn't prevent creating zig-zag routes using two distinct channels between peers. Having multiple public channels still seems to be quite common, also between bigger nodes. Some examples are shown in https://twitter.com/joostjgr/status/1309161962312531969/photo/1

That means that these channels are still easy to perform liquidity attacks on.

Detecting circular routes based on pubkey has been discussed in the previous PR and would be a solution: https://github.com/lightningnetwork/lnd/pull/3915#discussion_r370049019

I'd like to ask for a reconsideration of this.

**Comment by Crypt-iQ** [comment]
My opinion is that lnd should be a thing that routes anything. And you can have your own knobs to turn that affect the outcome.

**Comment by joostjager** [comment]
Yes, that is another way to look at it. That would even call for reverting the current protection of #3915. I think there is something to say for providing some protection by default, but of course users can use the htlc interceptor to achieve the same result.

**Comment by Crypt-iQ** [comment]
I think exposing tune-able knobs is our best bet here so we can expose control to the user without introducing code that may become debt in the future. WDYT?

**Comment by joostjager** [comment]
What does that look like exactly, those tune-able knobs? Command-line flags to allow or disallow circular routing?

**Comment by carlaKC** [comment]
> expose control to the user without introducing code that may become debt in the future

Isn't this just the htlc interceptor though? 

I think it makes sense to have some built in protections against forwards which are clearly fishy (eg, #3915 because there really isn't a good reason to circle through a single channel now that keysend is becoming more common), because not everybody will run a htlc interceptor. 

My reluctance with by-pubkey detection would still be that it disrupts circular rebalances, which people do operationally rely on. Although I'm not sure how common it is to do so with the same peer - perhaps we could try to get some idea of this on slack/twitter? If it's not going to be too much of a disruption, and is off by default, I think it's a reasonable addition.
